src/core/MeshBVH.js

```diff
@@ -211,7 +211,7 @@ export class MeshBVH {
 	}
 
 	/* Core Cast Functions */
-	raycast( ray, materialOrSide = FrontSide ) {
+	raycast( ray, materialOrSide = FrontSide, near = 0, far = Infinity ) {
```

Comment:
I'm wondering if it makes sense to just modify the interface here to take a `ray` or a `raycaster` object since the near and far values are already present on the raycaster. What do you think? Then we just pass a raycaster down through the internal functions instead of a ray.

I'll have to give this a little more thought.

<!--
It looks like the internal `checkIntersection` function in three.js that the one in this project is based on has changed to take a raycaster, as well, so it may make sense to synchronize the implementations again:

https://github.com/mrdoob/three.js/blob/a686f406b3092aa3217437899a027323db4a1f57/src/objects/Mesh.js#L333-L362
-->

Reply:
The code would definitely be cleaner